File: BarNone/TheBar/src/App/Data/Models/Comment/Comment.ts
import Lift from "App/Data/Models/Lift/Lift";

export default class Comment {
	public id: number;

	public sentUserID: number;

	public liftID: number;

	public timeSent: string;

	public text: string;

	public details?: CommentDetails;
}

class CommentDetails {
	public lift: Lift;
}

File: BarNone/Shared/BarNone.Shared.DomainModel/Comment/Comment.cs
using BarNone.Shared.DomainModel.Core;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace BarNone.Shared.DomainModel
{
    [Table("Comment", Schema = "public")]
    public class Comment : IDomainModel<Comment>, IOwnedDomainModel
    {
        [Key]
        public int ID { get; set; }

        public int UserID { get; set; }

        public int LiftID { get; set; }

        public DateTime TimeSent { get; set; }

        public string Text { get; set; }

        [ForeignKey("LiftID")]
        public Lift Lift { get; set; }

        [ForeignKey("UserID")]
        public User User { get; set; }
    }
}
